scilpy/io/streamlines.py

```diff
@@ -144,3 +144,33 @@ def load_tractogram_with_reference(parser, args, filepath,
         parser.error('{} is an unsupported file format'.format(filepath))
 
     return sft
+
+
+def filter_tractogram_data(tractogram, streamline_ids):
+    """ Filter tractogram data (data_per_streamline, data_per_point)
+    according to streamline ids
+
+    Parameters:
+    -----------
+    tractogram: Tractogram or StatefulTractogram
+        Tractogram containing the data to be filtered
+    streamline_ids: array_like
+        List of streamline ids the data corresponds to
+
+    Returns:
+    --------
+    A tuple containing
+        Dict of filtered data_per_streamline
+        Dict of filtered data_per_point
+    """
+
+    new_data_per_streamline = {}
+    new_data_per_point = {}
+    for key in tractogram.data_per_streamline.keys():
+        new_data_per_streamline[key] = \
+            [tractogram.data_per_streamline[key][i] for i in streamline_ids]
+    for key in tractogram.data_per_point.keys():
+        new_data_per_point[key] = \
+            [tractogram.data_per_point[key][i] for i in streamline_ids]
+
+    return new_data_per_streamline, new_data_per_point
```

Comment:
Since this can be done without a for loop, why not return streamlines too ?
(I mean outside of the current modification, in general this would be useful)

Reply:
You mean changing the function so it takes in a tractogram (either `Tractogram` or `StatefulTractogram`) and it returns a new tractogram with the selected streamlines and associated data ? Sure !

Reply:
yeah so input is covered fine, I think if the return is of the same type as the input, that would be perfect

Reply:
@frheault @EmmaRenauld should we only handle SFTs or Tractogram also ?

Reply:
I asked the question ~2 weeks ago on slack. People answered that SFT only is fine.

Reply:
Alright !